### PR TITLE
chore: retry when download templates script failed

### PR DESCRIPTION
**Pull request by hund030**
CD pipeline sometimes failed for download templates failed: https://github.com/OfficeDev/TeamsFx/actions/runs/3096650638/jobs/5012444613
Add retry logic to improve stability.

### CODE REVIEW

File: packages/fx-core/scripts/download-templates-zip.js [hunk 1]
```diff
@@ -22,6 +22,29 @@ async function step(desc, fn) {
   }
 }
 
+async function retry(id, fn, retryIntervalInMs = 5000, maxAttemptCount = 3) {
+  let exception = undefined;
+  for (let attempted = 0; attempted < maxAttemptCount; ++attempted) {
+    try {
+      if (attempted > 0) {
+        await sleep(retryIntervalInMs);
```
Comment (wenytang-ms) [inline packages/fx-core/scripts/download-templates-zip.js:30]: maybe each fail can make the retryIntervalInMs longer

Comment (hund030) [reply packages/fx-core/scripts/download-templates-zip.js]: Updated.

**Comment by wenytang-ms** [comment]
The business logic is fine, but from the JavaScript language point of view, it will not feel very beautiful.
Maybe you can ref this doc to make it better!
https://javascript.plainenglish.io/lets-make-a-retry-mechanism-a339307d44bc
https://solutional.ee/blog/2020-11-19-Proper-Retry-in-JavaScript.html

**Comment by hund030** [comment]
> The business logic is fine, but from the JavaScript language point of view, it will not feel very beautiful. Maybe you can ref this doc to make it better! https://javascript.plainenglish.io/lets-make-a-retry-mechanism-a339307d44bc https://solutional.ee/blog/2020-11-19-Proper-Retry-in-JavaScript.html

Updated. Thanks for the suggestions~

**Review by wenytang-ms — APPROVED** [review]

**Pull request merged**